### ISSUE TITLE
I cant visualize the output of an intermediate layer

### ISSUE THREAD

**Issue by lqj1990** [fchollet/keras#2684]
when I try to visualize the output of an intermediate layer which is mentioned in [http://keras.io/getting-started/faq/#how-can-i-visualize-the-output-of-an-intermediate-layer](http://keras.io/getting-started/faq/#how-can-i-visualize-the-output-of-an-intermediate-layer), I get a mistake
![image](https://cloud.githubusercontent.com/assets/4352933/15146591/ee59599e-16ee-11e6-945c-d19574acc281.png) I don‘t know what caused the problem. Maybe the instruction is not suitable for non-sequential model because it works well  on others' sequential model. Does any one encounter the same problem?

**Comment by jonasrla** [comment]
Are you sure you meant "get_3rd_layer_output = K.function([mode.layers[0].input], model.layers**[-1]**.output)"?

If that's not the problem, please show what you are trying to run.

**Comment by lqj1990** [comment]
@jonasrla Thank you for your answer.

Yes, you're right. I build a neural network, but the train loss, train accuracy, validate loss and validate accuracy are always unchanged. So I decided to use the function to see the output. 
`get_3rd_layer_output = K.function([model.layers[0].input],[model.layers[-1].output]);
output = get_3rd_layer_output([X_train])[0];`
But it still print the error, so I can't get this. 

I build a two-input neural network. The output after merging these two network will be input into a CNN and then use Softmax to do classification.

**Comment by lqj1990** [comment]
I've already solved the problem. (●'◡'●) I use the wrong layer of my model.
